_modules/hubble.py

```diff
@@ -135,6 +133,91 @@ def audit(configs=None,
     configs = [os.path.join(os.path.sep, os.path.join(*(con.split('.yaml')[0]).split('.')))
                for con in configs]
 
+    ret = _run_audit(configs, tags)
+
+    terse_results = {}
+    verbose_results = {}
+
+    # Pull out just the tag and description
+    terse_results['Failure'] = []
+    tags_descriptions = set()
+
+    for tag_data in ret.get('Failure', []):
+        tag = tag_data['tag']
+        description = tag_data.get('description')
+        if (tag, description) not in tags_descriptions:
+            terse_results['Failure'].append({tag: description})
+            tags_descriptions.add((tag, description))
+
+    terse_results['Success'] = []
+    tags_descriptions = set()
+
+    for tag_data in ret.get('Success', []):
+        tag = tag_data['tag']
+        description = tag_data.get('description')
+        if (tag, description) not in tags_descriptions:
+            terse_results['Success'].append({tag: description})
+            tags_descriptions.add((tag, description))
+
+    if not show_success and 'Success' in terse_results:
+        terse_results.pop('Success')
+
+    terse_results['Controlled'] = []
+    control_reasons = set()
+
+    for tag_data in ret.get('Controlled', []):
+        tag = tag_data['tag']
+        control_reason = tag_data.get('control', '')
+        description = tag_data.get('description')
+        if (tag, description, control_reason) not in control_reasons:
+            terse_results['Controlled'].append({tag: control_reason})
+            control_reasons.add((tag, description, control_reason))
+
+    if not terse_results['Controlled']:
+        terse_results.pop('Controlled')
+
+    # Format verbose output as single-key dictionaries with tag as key
+    if verbose:
+        verbose_results['Failure'] = []
+
+        for tag_data in ret.get('Failure', []):
+            tag = tag_data['tag']
+            verbose_results['Failure'].append({tag: tag_data})
+
+        verbose_results['Success'] = []
+
+        for tag_data in ret.get('Success', []):
+            tag = tag_data['tag']
+            verbose_results['Success'].append({tag: tag_data})
+
+        if not show_success and 'Success' in verbose_results:
+            verbose_results.pop('Success')
+
+        verbose_results['Controlled'] = []
+
+        for tag_data in ret.get('Controlled', []):
+            tag = tag_data['tag']
+            verbose_results['Controlled'].append({tag: tag_data})
+
+        if not verbose_results['Controlled']:
+            verbose_results.pop('Controlled')
+
+        results = verbose_results
+    else:
+        results = terse_results
+
+    if show_compliance:
```

Comment:
We need to calculate compliance before we `pop` the successes if `show_success` is False. Otherwise it will basically always report 0% compliance.